test/envoye2e/stats_plugin/stats_test.go

```diff
@@ -365,3 +365,37 @@ func TestAttributeGen(t *testing.T) {
 		})
 	}
 }
+
+func TestStatsParserRegression(t *testing.T) {
+	// This is a regression test for https://github.com/envoyproxy/envoy-wasm/issues/497
+	params := driver.NewTestParams(t, map[string]string{
+		"StatsFilterCode":         "inline_string: \"envoy.wasm.stats\"",
+		"WasmRuntime":             "envoy.wasm.runtime.null",
+		"StatsConfig":             driver.LoadTestData("testdata/bootstrap/stats.yaml.tmpl"),
+		"ClientHTTPFilters":       driver.LoadTestData("testdata/filters/stats_outbound.yaml.tmpl"),
+		"StatsFilterClientConfig": "{}",
+	}, envoye2e.ProxyE2ETests)
+	listener0 := params.LoadTestData("testdata/listener/client.yaml.tmpl")
+	params.Vars["StatsFilterClientConfig"] = driver.LoadTestJSON("testdata/stats/client_config_customized.yaml.tmpl")
+	listener1 := params.LoadTestData("testdata/listener/client.yaml.tmpl")
```

Comment:
So a simple filter reload would cause this?

Reply:
Yes. But it has to load without ANTLR first, then with it. The sequence is important.